internal/pluginframework/tfschema/struct_to_schema.go

```diff
@@ -0,0 +1,145 @@
+package tfschema
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/databricks/terraform-provider-databricks/internal/tfreflect"
+	dataschema "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func typeToSchema(v reflect.Value) map[string]AttributeBuilder {
+	scm := map[string]AttributeBuilder{}
+	rk := v.Kind()
+	if rk == reflect.Ptr {
+		v = v.Elem()
+		rk = v.Kind()
+	}
+	if rk != reflect.Struct {
+		panic(fmt.Errorf("schema value of Struct is expected, but got %s: %#v", rk.String(), v))
+	}
+	fields := tfreflect.ListAllFields(v)
+	for _, field := range fields {
+		typeField := field.StructField
+		fieldName := typeField.Tag.Get("tfsdk")
+		if fieldName == "-" {
+			continue
+		}
+		isOptional := fieldIsOptional(typeField)
+		kind := typeField.Type.Kind()
+		if kind == reflect.Ptr {
+			elem := typeField.Type.Elem()
+			sv := reflect.New(elem).Elem()
+			nestedScm := typeToSchema(sv)
+			scm[fieldName] = SingleNestedAttributeBuilder{Attributes: nestedScm, Optional: isOptional, Required: !isOptional}
+		} else if kind == reflect.Slice {
```

Comment:
```suggestion
		value := field.Value
		if kind == reflect.Ptr {
			kind = typeField.Type.Elem().Kind()
			value := field.Elem()
		}
		if kind == reflect.Slice {
```
Then on line 83 we use this `value` instead of `field.Value`. This means that pointers always receive consistent treatment as if they were not pointers from the perspective of schema calculation. It's basically the equivalent to lines 17-20.